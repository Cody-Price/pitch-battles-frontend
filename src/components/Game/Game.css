.game-main {
  overflow: hidden;
  height: 100vh;
}

.game-area-wrapper {
  background: url(./giantbg.png);
  max-width: 1500px;
  height: 750px;
  background-repeat: no-repeat;
  background-size: cover;
  image-rendering: pixelated;
  position: relative;
  margin: auto;
  overflow: hidden;
}

.game-header {
  position: relative;
  display: flex;
  justify-content: space-between;
  height: 82px;
}

.avatar-border {
  position: absolute;
  background: rgba(0, 0, 0, 0.7);
  height: 80px;
  width: 80px;
  margin: 1rem 0 0 1rem;
  display: flex;
  align-items: center;
  border-radius: 50%;
}

.avatar-border:hover {
  cursor: pointer;
}

.header-avatar {
  margin: auto;
  height: 62px;
  width: 37px;
}

.header-avatar.one {
  background: url(./pcs/char1/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.header-avatar.two {
  background: url(./pcs/char2/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.header-avatar.three {
  background: url(./pcs/char3/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.header-avatar.four {
  background: url(./pcs/char4/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.header-avatar.five {
  background: url(./pcs/char5/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.header-avatar.six {
  background: url(./pcs/char6/stand.png);
  image-rendering: pixelated;
  background-size: cover;
  background-repeat: no-repeat;
}

.hearts-and-timer {
  display: grid;
  grid-template-columns: 1fr 300px 1fr;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100px;
}

.timer {
  font-family: "chronotype";
  font-size: 3rem;
  line-height: 50px;
  padding-top: 0.6rem;
  margin-bottom: 0;
  margin-top: 0;
}

.timer-frame {
  background: url("./clockframe.png");
  image-rendering: pixelated;

  background-size: cover;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 3rem;
  width: 240px;
  height: 60px;
  margin: auto;
}

.gameplay-frame {
  position: relative;
  bottom: 0;
  left: 0;
  right: 0;
  margin-left: auto;
  margin-right: auto;
  width: 300px;
  height: 250px;
}
