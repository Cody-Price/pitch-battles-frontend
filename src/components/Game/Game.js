import React, { Component } from "react";

import Player from "../Player/Player";
import Monster from "../Monster/Monster";
import Staff from "../Staff/Staff";
import MonsterEffect from "../MonsterEffect/MonsterEffect";
import PlayerEffect from "../PlayerEffect/PlayerEffect";
import GameUserModal from "../GameUserModal/GameUserModal";
import Hearts from "../Hearts/Hearts";
import "./Game.css";

import avatars from "../../utilities/avatars";

class Game extends Component {
  constructor() {
    super();

    this.state = {
      currentLevel: 2,
      userModal: false,
      playerHearts: [0, 1, 2],
      monsterHearts: ["a", "b", "c", "d", "e"],
      currentTime: 0,
      start: 0,
      running: false,
      milliseconds: 0,
      playerHit: false,
      monsterHit: false
    };
  }

  componentDidMount() {
    this.startTimer();
  }

  toggleUserModal = () => {
    this.setState({
      userModal: !this.state.userModal
    });
  };

  startTimer() {
    this.setState({
      currentTime: this.state.currentTime / 1000,
      start: Date.now() - this.state.currentTime,
      isOn: true
    });
    this.timer = setInterval(
      () =>
        this.setState({
          currentTime: Math.ceil((Date.now() - this.state.start) / 1000),
          milliseconds: Date.now() - this.state.start
        }),
      500
    );
  }

  render() {
    const avatar = "1";
    return (
      <main className="game-main">
        <div className="game-area-wrapper">
          <header className="game-header">
            <div className="avatar-border" onClick={this.toggleUserModal}>
              <div className={`header-avatar ${avatars[avatar]}`} />
            </div>
            <GameUserModal status={this.state.userModal} />
            <section className="hearts-and-timer">
              <Hearts
                hearts={this.state.playerHearts}
                character="player-hearts"
              />
              <div className="timer-frame">
                <h3 className="timer">{this.state.currentTime}</h3>
              </div>
              <Hearts
                hearts={this.state.monsterHearts}
                character="monster-hearts"
              />
            </section>
          </header>
          <section className="gameplay-frame">
            <Player avatar={avatar} status="idle" />
            {this.state.playerHit && <PlayerEffect />}
            <Monster level={this.state.currentLevel} status="idle" />
            {this.state.monsterHit && <MonsterEffect />}
          </section>
          <Staff />
        </div>
      </main>
    );
  }
}

export default Game;
