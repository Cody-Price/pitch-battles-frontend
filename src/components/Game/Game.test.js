import React from "react";
import Game from "./Game";
import { shallow } from "./enzyme";

describe("Game", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Game />);
  });
});

it("is a test", () => {});
