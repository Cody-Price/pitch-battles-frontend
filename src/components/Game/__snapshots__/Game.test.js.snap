// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should match the snapshot 1`] = `
<main
  className="game-main"
>
  <div
    className="game-area-wrapper"
  >
    <header
      className="game-header"
    >
      <div
        className="game-avatar-circle"
        onClick={[Function]}
      />
      <div
        className="avatar-border"
      >
        <div
          className="header-avatar nine"
        />
      </div>
      <GameUserModal
        endGame={[Function]}
        instrument="flute"
        reset={[Function]}
        status={false}
      />
      <section
        className="hearts-and-timer"
      >
        <Hearts
          character="player-hearts"
          hearts={
            Array [
              0,
              1,
              2,
            ]
          }
        />
        <div
          className="timer-frame"
        >
          <h3
            className="timer"
          >
            -
          </h3>
        </div>
        <Hearts
          character="monster-hearts"
          hearts={Array []}
        />
      </section>
    </header>
    <section
      className="gameplay-frame"
    >
      <Player
        avatar={9}
        status="idle"
      />
      <Monster
        level={1}
        status="idle"
      />
    </section>
    <button
      className="kickoff-btn false"
      onClick={[Function]}
    >
      start
    </button>
    <Staff
      clef="treble"
      correct="idle"
      incorrect="idle"
      submitGuess={[Function]}
    />
  </div>
</main>
`;
